src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/DefaultRazorProjectService.cs

```diff
@@ -119,6 +119,16 @@ void AddDocumentToProject(IProjectSnapshot projectSnapshot, string textDocumentP
 
             _logger.LogInformation("Adding document '{filePath}' to project '{projectSnapshotFilePath}'.", filePath, projectSnapshot.FilePath);
             _projectSnapshotManagerAccessor.Instance.DocumentAdded(projectSnapshot.Key, hostDocument, textLoader);
+
+            // Adding a document to a project could also happen because a target was added to a project, or we're moving a document
+            // from Misc Project to a real one, and means the newly added document could actually already be open.
+            // If it is, we need to make sure we start generating it so we're ready to handle requests that could start coming in.
+            if (_projectSnapshotManagerAccessor.Instance.IsDocumentOpen(textDocumentPath) &&
```

Comment:
Consider capturing `_projectSnapshotManagerAccessor.Instance` in a local, since it's used here a couple of times and above.